### PR TITLE
Filtering by full text search

### PR DESCRIPTION
**Pull request by index-git**
Part of issue #257

- [x] Tests
- [x] Layman Test Client
- [x] Changlelog
- [x] Documentation

Optionally
- Add dependency to doc/dependencies.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ### Migrations and checks
 
 ### Changes
+- [#273](https://github.com/jirik/layman/issues/273) Endpoints [GET Layers](doc/rest.md#get-layers) and [GET Layers](doc/rest.md#get-maps) can filter and reorder results according to new query parameters.
```
Comment (index-git) [inline CHANGELOG.md:10]: `and [GET Layers]` -> `and [GET Maps]`

File: src/layman/common/rest.py [hunk 2]
```diff
@@ -98,3 +98,23 @@ def setup_post_access_rights(request_form, kwargs, actor_name):
         else:
             access_rights = list({x.strip() for x in request_form['access_rights.' + type].split(',')})
         kwargs['access_rights'][type] = access_rights
+
+
+def get_publications(publication_type, user):
+    map_infos_whole = layman_util.get_publication_infos(publ_type=publication_type,
```
Comment (index-git) [inline src/layman/common/rest.py:116]: map_infos_whole -> publication_infos_whole

File: src/layman/common/rest.py [hunk 3]
```diff
@@ -98,3 +98,23 @@ def setup_post_access_rights(request_form, kwargs, actor_name):
...
+                                                        context={'actor_name': user,
+                                                                 'access_type': 'read',
+                                                                 })
+
+    infos = [
+        {
+            'name': name,
+            'workspace': workspace,
+            'title': info.get("title", None),
```
Comment (index-git) [inline src/layman/common/rest.py:129]: Remove "None"

File: src/layman/layer/rest_layers.py [hunk 4]
```diff
@@ -22,19 +22,4 @@ def get():
     app.logger.info(f"GET Layers, user={g.user}")
 
     user = g.user.get('username') if g.user else settings.ANONYM_USER
```
Comment (index-git) [inline src/layman/layer/rest_layers.py:24]: get_current_user -> get_authn_username

File: src/layman/rest_multipublication_test.py [hunk 5]
```diff
@@ -83,16 +94,37 @@ def provide_data(self):
...
     @pytest.mark.parametrize('publication_type', process_client.PUBLICATION_TYPES)
     @pytest.mark.usefixtures('liferay_mock', 'ensure_layman', 'provide_data')
-    def test_get_publications(self, publication_type, headers, expected_publications):
-        infos = process_client.get_publications(publication_type, headers, )
-        info_publications = [(info.get('workspace'), info.get('name')) for info in infos]
-        assert expected_publications == info_publications
+    def test_get_publications(self, publication_type, headers, query_params, expected_publications):
+        infos = process_client.get_publications(publication_type, headers, query_params=query_params)
+        info_publications = [(info['workspace'], info['name']) for info in infos]
+        if isinstance(expected_publications, set):
```
Comment (index-git) [inline src/layman/rest_multipublication_test.py:127]: Rewrite to always test order.

**Review by jirik — DISMISSED** [review]

**Review by jirik — APPROVED** [review]

**Pull request merged**